python-lib/dku_config/stl_config.py

```diff
@@ -52,116 +57,44 @@ def _load_advanced_parameters(self, config):
+                        "cond": are_keys_in(
+                            ["trend", "low_pass", "seasonal_deg", "trend_deg", "low_pass_deg", "seasonal_jump", "trend_jump", "low_pass_jump"],
+                            additional_parameters),
+                        "err_msg": "The keys should be in the following iterable: ['trend', 'low_pass', 'seasonal_deg', "
+                                   "'trend_deg','low_pass_deg', 'seasonal_jump', 'trend_jump', 'low_pass_jump']."
                     }
                 ],
                 required=False
             )
-
-        additional_smoothers = config.get("stl_smoothers_kwargs")
-
-        if additional_smoothers:
-            are_smoothers_keys_valid = are_keys_in(["trend", "low_pass"], additional_smoothers)
-            minimum = max(self.period, 3)
-            are_smoothers_values_valid = all((x == "") or
-                                             ((is_odd(x) and float(x) > minimum)) for x in
-                                             additional_smoothers.values())
-
-            if are_smoothers_keys_valid and are_smoothers_values_valid:
-                smoothers_parsed = parse_values_to_int(additional_smoothers)
-            else:
-                smoothers_parsed = additional_smoothers
-
-            self.add_param(
-                name="additional_smoothers",
-                value=smoothers_parsed,
-                checks=[
-                    {
-                        "type": "is_type",
-                        "op": dict
-                    },
-                    {
-                        "type": "custom",
-                        "cond": are_smoothers_keys_valid,
-                        "err_msg": "This field is invalid. The keys should be in the following iterable: [trend, low_pass]."
-                    },
-                    {
-                        "type": "custom",
-                        "cond": are_smoothers_values_valid,
-                        "err_msg": f"This field is invalid. The values should be odd positive integers greater than 3 and the period (= {self.period})."
-                    }
-                ],
-                required=False
-            )
-
-
-def parse_values_to_int(map_parameter):
-    map_parameter_parsed = {}
-    for key, value in map_parameter.items():
-        if value:
-            map_parameter_parsed[key] = int(value)
-    return map_parameter_parsed
-
-
-def is_odd(x):
-    if is_positive_int(x):
-        numeric_value = int(x)
+            self._check_additional_parameters(additional_parameters)
+
+    def _check_additional_parameters(self, advanced_params):
+        for parameter_name, value in advanced_params.items():
+            additional_parameter = get_additional_parameter(parameter_name, value)
+            if not additional_parameter.is_valid(self):
+                raise ValueError(additional_parameter.get_full_error_message())
+
+
+def get_additional_parameter(parameter_name, value):
```

Comment:
This method could be inside the class (a class method for example) because it's only used within the class

Reply:
done in https://github.com/dataiku/dss-plugin-timeseries-preparation/pull/33/commits/918ea954842bd2891d54f256e885d15b63a19459